vyper/venom/basicblock.py

```diff
@@ -309,12 +341,108 @@ def is_pseudo(self) -> bool:
         """
         return self.is_phi or self.is_param
 
-    def get_read_effects(self):
+    def get_read_effects(self) -> effects.Effects:
         return effects.reads.get(self.opcode, effects.EMPTY)
 
-    def get_write_effects(self):
+    def get_write_effects(self) -> effects.Effects:
         return effects.writes.get(self.opcode, effects.EMPTY)
 
+    def get_write_memory_location(self) -> MemoryLocation:
+        """Extract memory location info from an instruction"""
+        opcode = self.opcode
+        if opcode == "mstore":
+            if isinstance(self.operands[1], IRLiteral):
+                return MemoryLocation(offset=self.operands[1].value, size=32)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "mload":
+            return EMPTY_MEMORY_ACCESS
+        elif opcode == "mcopy":
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[2].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "calldatacopy":
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[2].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "dloadbytes":
+            return FULL_MEMORY_ACCESS
+        elif opcode == "dload":
+            return FULL_MEMORY_ACCESS
+        elif opcode == "invoke":
+            return FULL_MEMORY_ACCESS
+        elif opcode in ("call", "delegatecall", "staticcall"):
+            if isinstance(self.operands[1], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(
+                    offset=self.operands[1].value, size=self.operands[0].value, is_volatile=False
+                )
+            return FULL_MEMORY_ACCESS
+        elif opcode in ("codecopy", "extcodecopy"):
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
+                return MemoryLocation(offset=self.operands[2].value, size=self.operands[0].value)
+            return FULL_MEMORY_ACCESS
+        elif opcode == "returndatacopy":
+            if isinstance(self.operands[2], IRLiteral) and isinstance(self.operands[0], IRLiteral):
```

Comment:
will be cleaner to deconstruct once --
```suggestion
            size, src, dst = self.operands
            if isinstance(dst, IRLiteral) and isinstance(size, IRLiteral):
                return MemoryLocation(offset=dst.value, size=size.value)
```